TelemetryEvents.swift

```diff
@@ -0,0 +1,45 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+
+private let PrefKeySearches = "Telemetry.Searches"
+
+public class SearchTelemetryFactory {
+    // For data consistency, the strings used here are identical to the ones reported in Android.
+    public enum Source: String {
+        case URLBar = "actionbar"
+        case QuickSearch = "listitem"
+        case Suggestion = "suggestion"
+    }
+
+    private init() {}
+
+    public class func makeEvent(engineID engineID: String, source: Source) -> TelemetryEvent {
+        return SearchTelemetryEvent(engineWithSource: "\(engineID).\(source.rawValue)")
```

Comment:
(Sorry, I updated this comment and you may have missed it)

I like how it encapsulates the appropriate code in their relevant places but it's lot harder to follow than just a call-through (e.g. using an IDE, if I'm looking into `Telemetry.recordEvent` or whatever, I see `event.record`, have to step back out of the function to jump to the declaration of the type `SearchEvent`, find the `record` method in that class (and have a good time trying to get back to `Telemetry.recordEvent` if I need it again). On the other hand, call throughs are just linear. If you have first class functions, you can still keep the code concise (but slightly less navigable) by declaring an enum -> function mapping).